config/settings.d/puppet.yml.example

```diff
@@ -41,16 +41,20 @@
 # If you want to override the puppet_user above just for mco commands
 #:mcollective_user: peadmin
 
-# URL of the puppet master itself for API requests
+# Override use of Puppet's API to list environments, by default it will use only if
+# environmentpath is given in puppet.conf, else will look for environments in puppet.conf
+# (Puppet versions prior to 4.0 only)
+#:puppet_use_environment_api: true
+
+# Puppet master version used
+#:puppet_version: 4.1
```

Comment:
There are three common scenarios I think we should handle:

1. Upgrades on a regular, existing Puppet 2/3 installation - could a migration detect the current Puppet version and fill that in?  Puppet should be require-able.
2. New Foreman installer installations, the installer will be templating this file and can supply the right value.
3. New installations, the config file could have a default value of the latest version of Puppet.  Perhaps we don't even need this one?